packages/notte-agent/src/notte_agent/falco/agent.py

```diff
@@ -336,8 +352,12 @@ async def _run(self, task: str, url: str | None = None) -> AgentResponse:
                             message=failed_val_msg,
                         )
                     )
+        finally:
```

Comment:
I don't like this. Let's remove that and ask the sessions to manage their sessions themselves